dexfile/src/main/kotlin/com/github/netomi/bat/dexfile/instruction/editor/InstructionBuilder.kt

```diff
@@ -65,6 +85,54 @@ class InstructionBuilder private constructor(private val dexEditor: DexEditor) {
         return invokeDirect(dexEditor.addOrGetMethodIDIndex(classType, name, parameterTypes, returnType), *registers)
     }
 
+    fun invokeVirtual(methodIndex: Int, vararg registers: Int): MethodInstruction {
+        return add(MethodInstruction.of(DexOpCode.INVOKE_VIRTUAL, methodIndex, *registers))
+    }
+
+    fun invokeVirtual(classType: String, name: String, parameterTypes: List<String>, returnType: String, vararg registers: Int): MethodInstruction {
+        return invokeVirtual(dexEditor.addOrGetMethodIDIndex(classType, name, parameterTypes, returnType), *registers)
+    }
+
+    fun invokeStatic(methodIndex: Int, vararg registers: Int): MethodInstruction {
+        return add(MethodInstruction.of(DexOpCode.INVOKE_STATIC, methodIndex, *registers))
+    }
+
+    fun invokeStatic(classType: String, name: String, parameterTypes: List<String>, returnType: String, vararg registers: Int): MethodInstruction {
+        return invokeStatic(dexEditor.addOrGetMethodIDIndex(classType, name, parameterTypes, returnType), *registers)
+    }
+
+    fun staticGetObject(classType: String, name: String, type: String, register: Int): FieldInstruction {
+        return add(FieldInstruction.of(DexOpCode.SGET_OBJECT, dexEditor.addOrGetFieldIDIndex(classType, name, type), register))
+    }
+
+    fun intToChar(registerA: Int, registerB: Int): ConversionInstruction {
```

Comment:
here we should better indicate what is the source and destination register.
comment can be applied likewise for the methods below.